lib/ansible/modules/network/cloudengine/ce_stp.py

```diff
@@ -0,0 +1,969 @@
+                        tmp_changed = True
+                    else:
+                        cmd = "undo stp root-protection"
+                        cmds.append(cmd)
+                        self.updates_cmd.append(cmd)
+                        tmp_changed = True
+
+            if self.loop_protection:
+                if self.loop_protection == "enable" and self.cur_cfg["root_protection"] == "enable":
+                    self.module.fail_json(
+                        msg='Error: The interface has enable root_protection, can not enable loop_protection.')
+                if self.loop_protection != self.cur_cfg["loop_protection"]:
+                    if self.loop_protection == "enable":
+                        cmd = "stp loop-protection"
+                        cmds.append(cmd)
+                        self.updates_cmd.append(cmd)
+                        tmp_changed = True
+                    else:
+                        cmd = "undo stp loop-protection"
+                        cmds.append(cmd)
+                        self.updates_cmd.append(cmd)
+                        tmp_changed = True
+
+            if self.cost:
+                if self.cost == self.cur_cfg["cost"]:
+                    cmd = "undo stp cost"
+                    cmds.append(cmd)
+                    self.updates_cmd.append(cmd)
+                    tmp_changed = True
+
+            if not tmp_changed:
+                cmd = "interface %s" % self.interface
+                self.updates_cmd.remove(cmd)
+                cmds.remove(cmd)
+
+        if cmds:
+            self.cli_load_config(cmds)
+            self.changed = True
+
+    def work(self):
+        """ Work function """
+
+        self.check_args()
+        self.get_proposed()
+        self.get_existing()
+
+        if self.state == "present":
+            self.present_stp()
+        else:
+            self.absent_stp()
+
+        self.get_end_state()
+
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        self.results['updates'] = self.updates_cmd
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """ Module main """
+
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        stp_mode=dict(choices=['stp', 'rstp', 'mstp']),
+        stp_enable=dict(choices=['enable', 'disable']),
+        stp_converge=dict(choices=['fast', 'normal']),
+        bpdu_protection=dict(choices=['enable', 'disable']),
+        tc_protection=dict(choices=['enable', 'disable']),
+        tc_protection_interval=dict(type='str'),
+        tc_protection_threshold=dict(type='str'),
+        interface=dict(type='str'),
+        edged_port=dict(choices=['enable', 'disable']),
+        bpdu_filter=dict(choices=['enable', 'disable']),
+        cost=dict(type='str'),
```

Comment:
For integers please:
use `type='int`
remove the `isdigit` check from `check_params`